consult.el

```diff
@@ -4763,38 +4829,52 @@ the directory to search in.  By default the project directory is used
 if `consult-project-function' is defined and returns non-nil.
 Otherwise the `default-directory' is searched."
   (interactive "P")
-  (consult--grep "Grep" (consult--grep-make-builder) dir initial))
+  ;; Must call `consult--directory-prompt' to set local `search-path-list'
+  ;; prior to calling `consult--grep-make-builder' where the path list is used.
+  (let* ((prompt-paths-dir (consult--directory-prompt "Grep" dir))
+	 (prompt (nth 0 prompt-paths-dir))
+	 (search-path-list (nth 1 prompt-paths-dir)))
+    (setq dir (nth 2 prompt-paths-dir))
+    (consult--grep prompt (consult--grep-make-builder search-path-list) dir initial)))
```

Comment:
It seems you replicated this code three times, for grep, git-grep and ripgrep. Instead you can move it into consult--grep and change consult--grep to take a MAKE-BUILDER argument instead of a BUILDER argument.

Furthermore I think we should reuse the DIR argument here. DIR can either be a single string which is then used as default-directory and searched or a list of strings for the wildcard handling.

Reply:
This should be resolved with next push.